etcd/etcd.go

```diff
@@ -304,7 +304,7 @@ func (obj *EmbdEtcd) GetConfig() etcd.Config {
 	// XXX: filter out any urls which wouldn't resolve here ?
 	for _, eps := range obj.endpoints { // flatten map
 		for _, u := range eps {
-			endpoints = append(endpoints, u.Host) // remove http:// prefix
+			endpoints = append(endpoints, u.String()) // use full url including scheme
```

Comment:
I feel like I took it off for a reason... Doesn't doing this break something elsewhere?

Reply:
AFAICT no tests are breaking. I tested this extensively last year when I started working on this feature and didn't find any issue. But there might be scenarios I'm not aware of?

Reply:
Can you try this one please:

https://github.com/purpleidea/mgmt/blob/90fa83a5cf28394617583578e982eeedd94a5628/examples/lang/exchange0.mcl#L1

Reply:
Test added: https://github.com/purpleidea/mgmt/pull/468/files#diff-34b136176f6a9b9c42585fa1ae2a07df

Reply:
Sweet! I was actually hoping that was it, but here's one more...
https://github.com/purpleidea/mgmt/blob/master/etcd/etcd.go#L38
I remember that code in the back of my mind, I just can't remember why it was necessary, but that it was. So it's really my bad for having all this old code... Which is why I'm nuking it shortly... In any case...

Reply:
I'm trying to convert this into a testcase as well, but on master this fails with:
```
~/p/m/mgmt (master|…) $ ./mgmt run --hostname h1 --tmp-prefix --no-pgp yaml --yaml examples/yaml/etcd1a.yaml
22:09:06 hello.go:49: this is: mgmt, version: 0.0.17-46-g4c80869
22:09:06 hello.go:50: main: start: 1550005746276466928
22:09:06 main.go:191: main: warning: working prefix directory is temporary!
22:09:06 main.go:191: main: working prefix is: /var/folders/00/wq2lzdvd6sg2w1_5l0092bj00000gn/T/mgmt-h1-776165371
22:09:06 main.go:191: main: etcd: seeds: no seeds specified!
22:09:06 etcd.go:435: Etcd: Bootstrapping...
22:09:06 etcd.go:350: Etcd: Connect: Endpoints: []
22:09:06 etcd.go:364: Etcd: Connect: CtxError...
22:09:06 etcd.go:644: Etcd: CtxError: Reason: CtxDelayErr(1s): No endpoints available yet!
22:09:06 etcd.go:1472: Etcd: Nominated: h1=http://localhost:2380
22:09:06 etcd.go:1487: Etcd: StartServer(newCluster: true): h1=http://localhost:2380
22:09:06 etcd.go:1758: Etcd: StartServer: Starting server...
22:09:06 etcd.go:1765: Etcd: StartServer: Done starting server!
22:09:06 etcd.go:1783: Etcd: StartServer: Server running...
22:09:06 etcd.go:1510: Etcd: Addresses are: http://localhost:2379
22:09:07 etcd.go:348: Etcd: Connect: Endpoints: h1=http://localhost:2379
22:09:07 etcd.go:1830: Etcd: ApplyDeltaEvents: Event(PUT): h1
22:09:07 etcd.go:1472: Etcd: Nominated: h1=http://localhost:2380
22:09:07 etcd.go:455: Etcd: Startup: Volunteering...
22:09:07 etcd.go:1678: Etcd: Ideal cluster size is now: 5
22:09:07 etcd.go:1022: Etcd: Set(/_mgmt/idealClusterSize): &{cluster_id:14841639068965178418 member_id:10276657743932975437 revision:4 raft_term:2  <nil>}
22:09:07 etcd.go:1272: Etcd: Members: List: [h1]
22:09:07 etcd.go:1295: Etcd: Leader: h1
22:09:07 etcd.go:1314: Etcd: Volunteers: [h1]
22:09:07 etcd.go:1318: Etcd: Quitters: []
22:09:07 etcd.go:1330: Etcd: Candidates: []
22:09:07 etcd.go:1272: Etcd: Members: List: [h1]
22:09:07 etcd.go:1295: Etcd: Leader: h1
22:09:07 etcd.go:1314: Etcd: Volunteers: [h1]
22:09:07 etcd.go:1318: Etcd: Quitters: []
22:09:07 etcd.go:1330: Etcd: Candidates: []
22:09:08 main.go:191: main: waiting...
22:09:08 main.go:191: main: running...
22:09:08 main.go:191: main: waiting...
22:09:08 main.go:492: gapi: generating new graph...
22:09:08 resources.go:41: Etcd: Watch: Path: /_mgmt/exported/
22:09:08 main.go:492: gapi: generating new graph...
22:09:08 resources.go:175: Etcd: Get: (Hostname, Kind, Name): (h1, file, filea)
22:09:08 main.go:492: gapi: yaml: collect: file; pattern: /tmp/mgmtA/
22:09:08 main.go:410: engine: autoedge: adding autoedges...
22:09:08 main.go:410: engine: autogroup: algorithm: wrappedGrouper: NonReachabilityGrouper...
22:09:08 main.go:191: main: commit...
22:09:08 main.go:410: engine: graph sync...
22:09:08 main.go:191: main: graph: Vertices(0), Edges(0)
22:09:08 main.go:191: main: waiting...
22:09:08 resources.go:175: Etcd: Get: (Hostname, Kind, Name): (h1, file, filea)
22:09:08 resources.go:109: Etcd: SetResources: Error: Empty kind: 
```

Reply:
Oh gah! That is so embarrassing of me, it's still a yaml example ;)
I told you this file was getting nuked =D

Just try the same example with the `yaml --yaml examples/yaml/etcd1a.yaml` replaced with `empty`

It's the cluster growing/shrinking volunteering/etc part that mattered, not the code that ran, sorry.

Thanks!

Reply:
Tests added :) https://github.com/purpleidea/mgmt/pull/468/files#diff-2fb98fd83d458bf7aeff523d04b46264